### PR TITLE
new release

### PR DESCRIPTION
**Pull request by guolinke**
- [x] complete #1637

### CODE REVIEW

**Comment by StrikerRUS** [comment]
@guolinke Pease take a look at https://github.com/Microsoft/LightGBM/pull/1629#issue-212222788.
I think it also should be resolved before new release.

**Comment by henry0312** [comment]
And we should check https://github.com/Microsoft/LightGBM/pull/1630.
Is v2.1.3 really right?
Is it sure that we follow [Semantic versioning](https://www.python.org/dev/peps/pep-0440/#semantic-versioning)?

**Comment by guolinke** [comment]
@henry0312 do you mean change the format of version?

**Comment by henry0312** [comment]
@guolinke yeah, I think we will release **v2.2.0** as next version if we follow [Semantic Versioning](https://semver.org/).

**Comment by guolinke** [comment]
@henry0312 why you think we need to jump a major version now?
@StrikerRUS what is your opinion?

**Comment by StrikerRUS** [comment]
To be honest, I agree with @henry0312 .

But I don't know what to do with the following code (list may be not completed)... Should we review these warnings?
https://github.com/Microsoft/LightGBM/blob/4a1e5fa59a4d45a6beab2e8039125bdfa7d12027/python-package/lightgbm/__init__.py#L39-L45
https://github.com/Microsoft/LightGBM/blob/4a1e5fa59a4d45a6beab2e8039125bdfa7d12027/python-package/lightgbm/basic.py#L2097-L2098
https://github.com/Microsoft/LightGBM/blob/4a1e5fa59a4d45a6beab2e8039125bdfa7d12027/python-package/lightgbm/sklearn.py#L578-L579
https://github.com/Microsoft/LightGBM/blob/4a1e5fa59a4d45a6beab2e8039125bdfa7d12027/python-package/lightgbm/plotting.py#L351-L352
https://github.com/Microsoft/LightGBM/blob/4a1e5fa59a4d45a6beab2e8039125bdfa7d12027/python-package/lightgbm/plotting.py#L427-L428
https://github.com/Microsoft/LightGBM/blob/4a1e5fa59a4d45a6beab2e8039125bdfa7d12027/src/objective/binary_objective.hpp#L54

**Comment by guolinke** [comment]
@StrikerRUS I think we should fix them accordingly.

**Comment by StrikerRUS** [comment]
@guolinke Do you have thoughts how exactly?

**Comment by guolinke** [comment]
@StrikerRUS my most concern is the one with apple clang. 

For the rest, we can according to the new version number to deal with them.

**Comment by StrikerRUS** [comment]
@guolinke 
>my most concern is the one with apple clang.

Change the version in the warning to 2.2.1 or 2.3?

**Comment by guolinke** [comment]
@StrikerRUS I think 2.2.1 is okay.

**Comment by StrikerRUS** [comment]
@guolinke Me too.
I think for `pred_parameter` and `apply method` we can change nothing and remove after 2.2 release like it's said in the warning. For `plotting kwargs` and `binary objective` we can change from 2.3. to 2.4, I suppose.

**Comment by henry0312** [comment]
that seems good to me.
https://github.com/Microsoft/LightGBM/pull/1649#issuecomment-420952334

**Comment by guolinke** [comment]
@StrikerRUS can you help to address these warnings in another PR ?

**Comment by StrikerRUS** [comment]
@guolinke Sure!

**Comment by guolinke** [comment]
yeah, it had tested it.

File: VERSION.txt [hunk 1]
```diff
@@ -1 +1 @@
-2.1.2.4
+2.2
```
Comment (henry0312) [inline VERSION.txt]: I think this should be **2.2.0**

**Comment by guolinke** [comment]
@StrikerRUS I think the docker are checked in the "tag", not release.

**Comment by StrikerRUS** [comment]
@guolinke Ah, seems you're right.

**Pull request merged**